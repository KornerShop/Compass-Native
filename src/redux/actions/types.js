export const SET_LANGUAGE_PREFERENCE = 'SET_LANGUAGE_PREFERENCE'
export const TOGGLE_LOCATION_PROVIDED = 'TOGGLE_LOCATION_PROVIDED'
export const UPDATE_ZIPCODE = 'UPDATE_ZIPCODE'
export const POPULATE_SNAP_OFFICES = 'POPULATE_SNAP_OFFICES'
export const POPULATE_WIC_OFFICES = 'POPULATE_WIC_OFFICES'
export const UPDATE_WIC_ELIGIBILITY = 'UPDATE_WIC_ELIGIBILITY'
export const UPDATE_ORIENTATION = 'UPDATE_ORIENTATION'
export const UPDATE_LOCATION = 'UPDATE_LOCATION'
export const UPDATE_OFFICE = 'UPDATE_OFFICE'
export const TOGGLE_MAP_LOADING = 'TOGGLE_MAP_LOADING'
