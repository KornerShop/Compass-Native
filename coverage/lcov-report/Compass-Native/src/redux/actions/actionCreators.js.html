<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Compass-Native/src/redux/actions/actionCreators.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">Compass-Native/src/redux/actions</a> actionCreators.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.29% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>33/35</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>28/30</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import moment from 'moment';
import {
  updateMapLoading,
  updateLocation,
  populateSNAP,
  populateWIC,
  setLanguagePreference,
  updateOffice,
  updateZipCode,
} from './actions';
&nbsp;
import {
  fetchZipCode,
  fetchZipCodeCoords,
  fetchResults,
} from '../../utilities/mapUtils';
&nbsp;
export const updateLanguage = (socket, lang) =&gt; dispatch =&gt; {
  socket.emit('update-language', {
    lang: lang === 'es' ? 'Spanish' : <span class="branch-1 cbranch-no" title="branch not covered" >'English',</span>
  });
  dispatch(setLanguagePreference(lang));
};
&nbsp;
export const changeOffice = (socket, office) =&gt; dispatch =&gt; {
  socket.emit('update-office', {
    office: office === 1 ? 'SNAP' : <span class="branch-1 cbranch-no" title="branch not covered" >'WIC',</span>
    date: moment().format('l')
  });
  dispatch(updateOffice(office));
};
&nbsp;
export const changeZipCode = (socket, zipCode) =&gt; dispatch =&gt; {
  socket.emit('update-zip', {
    zipCode,
  });
  dispatch(updateZipCode(zipCode));
};
&nbsp;
export const changeLocation = (socket, location) =&gt; async dispatch =&gt; {
  dispatch(updateLocation(location))
  const zipCode = await fetchZipCode(location);
  socket.emit('update-zip', {
    zipCode,
  });
};
&nbsp;
export const updateOffices = async (
  dispatch,
  officeNum,
  latitude,
  longitude,
) =&gt; {
  let keyword;
  let action;
  <span class="missing-if-branch" title="else path not taken" >E</span>if (officeNum === 1) {
    keyword = 'calfresh';
    action = populateSNAP;
  } else {
<span class="cstat-no" title="statement not covered" >    keyword = 'wic';</span>
<span class="cstat-no" title="statement not covered" >    action = populateWIC;</span>
  }
  const offices = await fetchResults(latitude, longitude, keyword);
  dispatch(action(offices));
  dispatch(updateMapLoading(false));
};
&nbsp;
export const fetchOffices = bool =&gt; async (dispatch, getState) =&gt; {
  dispatch(updateMapLoading(true));
  const { office, zipCode } = getState();
  if (bool) {
    var { lat: latitude, lng: longitude } = await fetchZipCodeCoords(
      zipCode,
    );
  } else {
    var { latitude, longitude } = getState().location;
  }
  dispatch(updateLocation({ latitude, longitude }));
  updateOffices(dispatch, office, latitude, longitude);
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jul 17 2017 07:56:03 GMT-0700 (PDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
