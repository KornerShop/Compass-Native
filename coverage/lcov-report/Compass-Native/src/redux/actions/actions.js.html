<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Compass-Native/src/redux/actions/actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">Compass-Native/src/redux/actions</a> actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>20/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as types from './types';
&nbsp;
export const setLanguagePreference = pref =&gt; ({
  type: types.SET_LANGUAGE_PREFERENCE,
  payload: pref,
});
&nbsp;
export const toggleLocationProvided = provided =&gt; ({
  type: types.TOGGLE_LOCATION_PROVIDED,
  payload: provided,
});
&nbsp;
export const updateZipCode = zipCode =&gt; ({
  type: types.UPDATE_ZIPCODE,
  payload: zipCode,
});
&nbsp;
export const updateWicEligibility = eligible =&gt; ({
  type: types.UPDATE_WIC_ELIGIBILITY,
  payload: eligible,
});
&nbsp;
export const updateOrientation = dimensions =&gt; ({
  type: types.UPDATE_ORIENTATION,
  payload: dimensions,
});
&nbsp;
export const updateOffice = office =&gt; ({
  type: types.UPDATE_OFFICE,
  payload: office,
});
&nbsp;
export const updateMapLoading = newState =&gt; ({
  type: types.TOGGLE_MAP_LOADING,
  payload: newState,
});
&nbsp;
export const updateLocation = location =&gt; ({
  type: types.UPDATE_LOCATION,
  payload: location,
});
&nbsp;
export const populateSNAP = offices =&gt; ({
  type: types.POPULATE_SNAP_OFFICES,
  payload: offices,
});
&nbsp;
export const populateWIC = offices =&gt; ({
  type: types.POPULATE_WIC_OFFICES,
  payload: offices,
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jul 17 2017 07:56:03 GMT-0700 (PDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
