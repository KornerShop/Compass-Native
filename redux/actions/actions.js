import {createAction} from 'redux-actions'

import * as types from './types'

export const setLanguagePreference = pref =>
  createAction(types.SET_LANGUAGE_PREFERENCE)(pref)

export const toggleLocationPermission = perm =>
  createAction(types.TOGGLE_LOCATION_PERMISSION)(perm)

export const updateZipcode = zip => createAction(types.UPDATE_ZIPCODE)(zip)

export const populateSnapOffices = offices =>
  createAction(types.POPULATE_SNAP_OFFICES)(offices)

export const populateWicOffices = offices =>
  createAction(types.POPULATE_WIC_OFFICES)(offices)

export const toggleWicEligibility = eligible =>
  createAction(types.TOGGLE_WIC_ELIGIBILITY)(eligible)
