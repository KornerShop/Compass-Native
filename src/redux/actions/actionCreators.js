import moment from 'moment';
import {
  updateMapLoading,
  updateLocation,
  populateSNAP,
  populateWIC,
  populateWICVendors,
  setLanguagePreference,
  updateOffice,
  updateZipCode,
  populateFoodBanks,
} from './actions';

import {
  fetchZipCode,
  fetchZipCodeCoords,
  fetchResults,
  fetchWICVendorsLocationPermission,
  fetchWICVendorsZipCode,
  fetchFoodBanks,
} from '../../utilities/mapUtils';

export const updateLanguage = (socket, lang) => dispatch => {
  socket.emit('update-language', {
    lang: lang === 'es' ? 'Spanish' : 'English',
  });
  dispatch(setLanguagePreference(lang));
};

export const changeOffice = (socket, office) => dispatch => {
  socket.emit('update-office', {
    office: office === 1 ? 'SNAP' : 'WIC',
    date: moment().format('l'),
  });
  dispatch(updateOffice(office));
};

export const changeZipCode = (socket, zipCode) => dispatch => {
  socket.emit('update-zip', {
    zipCode,
  });
  dispatch(updateZipCode(zipCode));
};

export const changeLocation = (
  socket,
  location,
) => async dispatch => {
  dispatch(updateLocation(location));
  const zipCode = await fetchZipCode(location);
  socket.emit('update-zip', {
    zipCode,
  });
};

export const updateOffices = () => async (dispatch, getState) => {
  dispatch(updateMapLoading(true));
  const { office, zipCode } = getState();
  if (zipCode) {
    var { lat: latitude, lng: longitude } = await fetchZipCodeCoords(
      zipCode,
    );
  } else {
    var { location: { latitude, longitude } } = getState();
  }
  dispatch(updateLocation({ latitude, longitude }));
  if (office === 1) {
    dispatch(
      populateSNAP(
        await fetchResults(latitude, longitude, 'calfresh'),
      ),
    );
  } else {
    dispatch(
      populateWIC(await fetchResults(latitude, longitude, 'wic')),
    );
  }
  dispatch(updateMapLoading(false));
};

export const updateWICVendorsZipModal = zipCode => async (
  dispatch,
  getState,
) => {
  const { wicOffices, snapOffices } = getState();
  dispatch(populateWICVendors(await fetchWICVendorsZipCode(zipCode)));
  wicOffices && snapOffices && dispatch(updateMapLoading(false));
};

export const updateWICVendorsLocationPermission = (
  latitude,
  longitude,
) => async (dispatch, getState) => {
  const { wicOffices, snapOffices } = getState();
  dispatch(
    populateWICVendors(
      await fetchWICVendorsLocationPermission(latitude, longitude),
    ),
  );
  wicOffices && snapOffices && dispatch(updateMapLoading(false));
};

export const getFoodBanks = async (dispatch, latitude, longitude) => {
  dispatch(updateLocation({ latitude, longitude }));
  const foodBanks = await fetchFoodBanks(latitude, longitude);
  dispatch(populateFoodBanks(foodBanks));
  dispatch(updateMapLoading(false));
};

export const updateFoodBanks = () => async (dispatch, getState) => {
  dispatch(updateMapLoading(true));
  const { zipCode } = getState();
  if (zipCode) {
    const {
      lat: latitude,
      lng: longitude,
    } = await fetchZipCodeCoords(zipCode);
    getFoodBanks(dispatch, latitude, longitude);
  } else {
    const { latitude, longitude } = getState().location;
    getFoodBanks(dispatch, latitude, longitude);
  }
};
