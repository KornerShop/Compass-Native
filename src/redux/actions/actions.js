import {createAction} from 'redux-actions'

import * as types from './types'

import getLayoutInfo from '../../utilities/orientation'

export const setLanguagePreference = pref =>
	createAction(types.SET_LANGUAGE_PREFERENCE)(pref)

export const toggleLocationPermission = perm =>
	createAction(types.TOGGLE_LOCATION_PERMISSION)(perm)

export const updateZipcode = zip => createAction(types.UPDATE_ZIPCODE)(zip)

export const populateSnapOffices = offices =>
	createAction(types.POPULATE_SNAP_OFFICES)(offices)

export const populateWicOffices = offices =>
	createAction(types.POPULATE_WIC_OFFICES)(offices)

export const updateWicEligibility = eligible =>
	createAction(types.UPDATE_WIC_ELIGIBILITY)(eligible)

export const updateOrientation = dimensions =>
	createAction(types.UPDATE_ORIENTATION)(getLayoutInfo(dimensions))

export const updateLocation = location => {
	createAction(types.UPDATE_LOCATION)(regionFrom(location))
}

export const updateOffice = office => createAction(types.UPDATE_OFFICE)(office)
