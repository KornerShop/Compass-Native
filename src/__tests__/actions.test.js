import * as actions from '../redux/actions/actions';

test('setLanguagePreference', () => {
  expect(actions.setLanguagePreference('es')).toMatchSnapshot();
});

test('toggleLocationProvided', () => {
  expect(actions.toggleLocationProvided(true)).toMatchSnapshot();
});

test('updateZipCode', () => {
  expect(actions.updateZipCode(95404)).toMatchSnapshot();
});

test('updateWicEligibility', () => {
  expect(actions.updateWicEligibility(1)).toMatchSnapshot();
});

test('updateOrientation', () => {
  expect(
    actions.updateOrientation({
      width: 411.42857142857144,
      height: 683.4285714285714,
      scale: 2.625,
      fontScale: 1,
    }),
  ).toMatchSnapshot();
});

test('updateOffice', () => {
  expect(actions.updateOffice(1)).toMatchSnapshot();
});

test('updateMapLoading', () => {
  expect(actions.updateMapLoading(false)).toMatchSnapshot();
});

test('updateLocation', () => {
  expect(
    actions.updateLocation({
      latitude: 38.5386881,
      longitude: -122.695547,
      latitudeDelta: 0.4,
      longitudeDelta: 0.4,
    }),
  ).toMatchSnapshot();
});

test('populateSNAP', () => {
  expect(
    actions.populateSNAP([
      {
        id: 'ChIJWcnjX3c4hIAReEaEIu-h5Dg',
        lat: 38.464769,
        lng: -122.721573,
        name: 'Sonoma County Human Services Department',
        address: '2550 Paulin Dr # 1, Santa Rosa, CA 95403, USA',
        phone_local: '(877) 699-6868',
        phone_intl: '+1 877-699-6868',
      },
      {
        id: 'ChIJ2S8f1PhHhIARUVCsEnyrOEw',
        lat: 38.443262,
        lng: -122.715587,
        name: 'Sonoma County Economic Assistance 2',
        address: '520 Mendocino Ave, Santa Rosa, CA 95402, USA',
        phone_local: '(877) 699-6868',
        phone_intl: '+1 877-699-6868',
      },
    ]),
  ).toMatchSnapshot();
});

test('populateWIC', () => {
  expect(
    actions.populateWIC([
      {
        id: 'ChIJ697sM6e2hYARGih3Xt7SAAE',
        lat: 38.2319379,

        lng: -122.6442755,
        name: 'WIC Petaluma',
        address: '11 English St, Petaluma, CA 94952, USA',
        phone_local: '(800) 816-3663',
        phone_intl: '+1 800-816-3663',
      },
      {
        id: 'ChIJ8RLqxDw4hIARCChYAI7Tp20',
        lat: 38.439528,
        lng: -122.7437988,
        name: 'Volunteer Center of Sonoma County',
        address: '153 Stony Cir #100, Santa Rosa, CA 95401, USA',
        phone_local: '(707) 573-3399',
        phone_intl: '+1 707-573-3399',
      },
      {
        id: 'ChIJD2NWug1zhYARmELjE9P_loU',
        lat: 38.1016731,
        lng: -122.2390356,
        name: 'Prime Time Nutrition',
        address: '1530 Solano Ave, Vallejo, CA 94590, USA',
        phone_local: '(707) 644-6600',
        phone_intl: '+1 707-644-6600',
      },
      {
        id: 'ChIJ24DSyww-hIAROpzUsj2K5Zs',
        lat: 38.5431579,
        lng: -122.8017116,
        name: 'Alliance Medical Center',
        address: '8465 Old Redwood Hwy #320, Windsor, CA 95492, USA',
        phone_local: '(707) 433-5494',
        phone_intl: '+1 707-433-5494',
      },
      {
        id: 'ChIJy1-q0DRzhYARuIqXldjxVYw',
        lat: 38.13150350000001,
        lng: -122.2387657,
        name: 'La Clinica Vallejo Great Begin',
        address: '210 Hospital Dr, Vallejo, CA 94589, USA',
        phone_local: '(707) 645-7316',
        phone_intl: '+1 707-645-7316',
      },
    ]),
  );
});
