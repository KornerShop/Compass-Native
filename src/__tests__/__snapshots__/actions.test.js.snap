// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`populateSNAP 1`] = `
Object {
  "payload": Array [
    Object {
      "address": "2550 Paulin Dr # 1, Santa Rosa, CA 95403, USA",
      "id": "ChIJWcnjX3c4hIAReEaEIu-h5Dg",
      "lat": 38.464769,
      "lng": -122.721573,
      "name": "Sonoma County Human Services Department",
      "phone_intl": "+1 877-699-6868",
      "phone_local": "(877) 699-6868",
    },
    Object {
      "address": "520 Mendocino Ave, Santa Rosa, CA 95402, USA",
      "id": "ChIJ2S8f1PhHhIARUVCsEnyrOEw",
      "lat": 38.443262,
      "lng": -122.715587,
      "name": "Sonoma County Economic Assistance 2",
      "phone_intl": "+1 877-699-6868",
      "phone_local": "(877) 699-6868",
    },
  ],
  "type": "POPULATE_SNAP_OFFICES",
}
`;

exports[`setLanguagePreference 1`] = `
Object {
  "payload": "es",
  "type": "SET_LANGUAGE_PREFERENCE",
}
`;

exports[`toggleLocationProvided 1`] = `
Object {
  "payload": true,
  "type": "TOGGLE_LOCATION_PROVIDED",
}
`;

exports[`updateLocation 1`] = `
Object {
  "payload": Object {
    "latitude": 38.5386881,
    "latitudeDelta": 0.4,
    "longitude": -122.695547,
    "longitudeDelta": 0.4,
  },
  "type": "UPDATE_LOCATION",
}
`;

exports[`updateMapLoading 1`] = `
Object {
  "payload": false,
  "type": "TOGGLE_MAP_LOADING",
}
`;

exports[`updateOffice 1`] = `
Object {
  "payload": 1,
  "type": "UPDATE_OFFICE",
}
`;

exports[`updateOrientation 1`] = `
Object {
  "payload": Object {
    "fontScale": 1,
    "height": 683.4285714285714,
    "scale": 2.625,
    "width": 411.42857142857144,
  },
  "type": "UPDATE_ORIENTATION",
}
`;

exports[`updateWicEligibility 1`] = `
Object {
  "payload": 1,
  "type": "UPDATE_WIC_ELIGIBILITY",
}
`;

exports[`updateZipCode 1`] = `
Object {
  "payload": 95404,
  "type": "UPDATE_ZIPCODE",
}
`;
